pkg/api/dirs.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package api
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"mime"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/manifest/jsonmanifest"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+// DirUploadResponse is returned when an HTTP request to upload a directory as a tar is successful
+type DirUploadResponse struct {
```

Comment:
Why not using same system as for the `bytesPostResponse` and `fileUploadResponse` structs, which have non-exported main struct, and special ones for test-only, in [export_test.go](https://github.com/ethersphere/bee/blob/master/pkg/api/export_test.go)?

Reply:
you're right, they are entirely the same.

i've removed this struct and changed the code so that it uses `fileUploadResponse`

Reply:
I had no objections to `DirUploadResponse` struct, just noted that it could be non-exported, and in `export_test.go` have:
```
DirUploadResponse = dirUploadResponse
```
so it can be used if needed.

Reply:
but do you think we really need a different response in this sense?

after all, it really _is_ a file upload response, specifically the manifest upload.